src/pkg/services/m365/api/pagers/pagers.go

```diff
@@ -410,6 +410,75 @@ type addedAndRemovedHandler[T any] func(
 	error,
 )
 
+func getLimitedItems[T any](
+	ctx context.Context,
+	resultsPager *nextPageResults[T],
+	itemLimit int,
+	getAddedAndRemoved addedAndRemovedHandler[T],
+	filters ...func(T) bool,
+) (map[string]time.Time, []string, DeltaUpdate, error) {
+	var (
+		done  bool
+		reset bool
+		page  []T
+
+		removed []string
+		added   = map[string]time.Time{}
+	)
+
+	// Can't use a for-loop variable declaration because the line ends up too
+	// long.
+	for !done && (itemLimit <= 0 || len(added) < itemLimit) {
```

Comment:
Whats the significance of `itemLimit <= 0`? Is this a way to do a complete enumeration?

Reply:
yup! since pagers are already used in a few places I was trying to avoid having to setup default options on the call config (e.x. use `-1` for "enumerate all"). Since 0 is the zero-value for int's that's what I was hoping to leverage in order to keep other changes smaller